Sources/MapboxCoreNavigation/CoreConstants.swift

```diff
@@ -292,6 +292,16 @@ extension RouteController {
          */
         public static let roadNameKey: NotificationUserInfoKey = .init(rawValue: "roadName")
         
+        /**
+         A key in the user info dictionary of a `Notification.Name.currentRoadNameDidChange` notification. The corresponding value is a `MapboxNavigationNative.Shield` object representing the road shield the user is currently traveling on.
```

Comment:
The notification user info dictionary is also used by application code. We shouldn’t store any C++ wrapper objects from MapboxNavigationNative directly in this dictionary, because these objects have different semantics than in pure Swift code. Instead, convert the shield object into a `VisualInstruction.Component.ShieldRepresentation` instance.

Reply:
This _might_ be the reason for the CocoaPods linker errors. MapboxNavigation.podspec doesn’t directly depend on MapboxNavigationNative, because the expectation is that anything vended by MapboxCoreNavigation can be used in pure Swift code without a separate MapboxNavigationNative dependency.

Reply:
Thanks for the suggestion, I'll update this part and to solve this issue.

Reply:
Yes, the error is solved with the `VisualInstruction.Component.ShieldRepresentation`  instead of the `NavigationStatus.Shield`.